cmd/gitlab.go

```diff
@@ -386,86 +362,145 @@ func (c *Client) pollPipelineJobs(p Project, pipelineID int, topics string, ref
 	return err
 }
 
-func (c *Client) pollProjectRef(p Project, ref string) {
-	var gp *gitlab.Project = p.GitlabProject
-	topics := strings.Join(gp.TagList[:], ",")
+func (c *Client) pollProjectRef(p Project, ref string) error {
+	var gp = p.GitlabProject
 	var lastPipeline *gitlab.Pipeline
+	topics := strings.Join(gp.TagList[:], ",")
 
+	//FIXME is adding the counter a 0 value at every run a problem?
```

Comment:
agreed, it would be good to handle this initialization a bit better

Reply:
I was not able to test it properly but it looks like this is not a problem, it's only re-setting that 0 value at every run.
I don't have enough knowledge on Prometheus library to understand if it's ok to keep it or not and could not construct a proper test for it...

Reply:
it does not reset the value to 0, it tries to add `0` which has the effect of initializing the metric on the first run and do not do anything on the subsequent ones.

Reply:
So basically has no harm on existing metrics? It will just create them the first time? If yes, we can leave it like that

Reply:
yes 👍